src/common/validators/scripture-references.validator.ts

```diff
@@ -0,0 +1,52 @@
+import { ValidationArguments, ValidationOptions } from 'class-validator';
+import { books } from '../../components/scripture/books';
+import { bookIndexFromName } from '../../components/scripture/reference';
+import { ValidateBy } from './validateBy';
+
+export const IsValidChapter = (
+  book: string,
+  validationOptions?: ValidationOptions
+) =>
+  ValidateBy(
+    {
+      name: 'isValidChapter',
+      constraints: [book],
+      validator: {
+        validate: (value, args: ValidationArguments) => {
+          const [book] = args.constraints;
+          const bookName = (args.object as any)[book];
```

Comment:
I think it's ok to assume the shape of the object. It will only be used on `ScriptureReference` object.

Reply:
Just pushed a new commit. Can you suggest the code change to assume the shape of the object?